models/articles.models.js

```diff
@@ -6,12 +6,25 @@ exports.fetchTopics = () => {
   });
 };
 
-exports.fetchArticleById = (articleId) => {
-  return db
-    .query(`SELECT * FROM articles WHERE article_id = $1`, [articleId])
-    .then((data) => {
-      return data.rows[0];
-    });
+exports.fetchArticleById = async (articleId) => {
+  const result1 = await db.query(
+    `SELECT articles.article_id 
+    FROM comments 
+    INNER JOIN articles
+    ON comments.article_id = articles.article_id
+    WHERE articles.article_id = $1;`,
+    [articleId]
+  );
+  const result2 = await db.query(
+    `SELECT * FROM articles WHERE article_id = $1;`,
+    [articleId]
+  );
+
+  const totalComments = result1.rows.length;
```

Comment:
unfortunately we shouldn't be using JS to count the comments here. We should be able to do it all in one query string. Have a see if you can refactor the query string to count the comments. Look into the COUNT aggregate function and you might need a LEFT JOIN instead of inner...